doc/computing/randomness.rst

```diff
@@ -0,0 +1,315 @@
+    0.8
+
+.. note::
+    `score()` is not a random procedure. Only `fit` has randomness.
+
+We can see from the snippet above that `rf.fit` has produced different
+models, even if the data was the same. This is because the RNG of the
+estimator is consumed when `fit` is called, and this consumed (mutaded) RNG
+will be used in the subsequent calls to `fit`. In addition, the `rng`
+object is shared across any object that uses it, and as a consequence, these
+object become somewhat inter-dependent. For example, two estimator that share
+the same `RandomState` instance will influence each-other, as we will see
+later when we discuss cloning.
+
+If we had passed an int to the `random_state` parameter of the
+:class:`~sklearn.ensemble.RandomForestClassifier`, we would have obtained the
+same models, and thus the same scores each time. When we pass an int, the
+same RNG is used across all calls to `fit`. What internally happens is that
+even though the RNG is consumed when `fit` is called, it is always reset to
+its original state at the beginning of `fit`.
+
+.. note::
+    Using `max_features=2, max_samples=10` is likely to be a poor choice in
+    general for a random forest. We here only set these parameters for
+    illustration purposes: otherwise the scores would not significantly
+    differ. The variance of the fitted models can typically be reduced by
+    increasing the number of randomly drawn samples and features, or by using
+    more trees.
+
+CV splitters
+............
+
+Randomized cv splitters have a similar behavior when a `RandomState`
+instance is passed::
+
+    >>> from sklearn.model_selection import KFold
+    >>> import numpy as np
+
+    >>> X = np.arange(10)
+    >>> rng = np.random.RandomState(0)
+    >>> cv = KFold(n_splits=2, shuffle=True, random_state=rng)
+    >>> for train, test in cv.split(X):
+    ...     print(train, test)
+    [0 3 5 6 7] [1 2 4 8 9]
+    [1 2 4 8 9] [0 3 5 6 7]
+    >>> for train, test in cv.split(X):
+    ...     print(train, test)
+    [0 4 6 7 8] [1 2 3 5 9]
+    [1 2 3 5 9] [0 4 6 7 8]
+
+We can see that the splits are different from the second time `split` is
+called. This may lead to unexpected results if you compare the performance of
+multiple estimators by calling `split` many times: the estimators will not
+be evaluated on the same folds, and the scores on each fold will not be
+comparable. On average, if enough folds are used and with enough data, one
+can however expect that that mean score allows to conclude whether one
+estimator is better than another. In any case, passing an int makes score
+comparisons across folds meaningful, and is usually safer.
+
+
+Common pitfalls and subtleties
+------------------------------
+
+While the rules that govern the `random_state` parameter are seemingly simple,
+they do however have some subtle implications. In some cases, this can even
+lead to wrong conclusions.
+
+Estimators
+..........
+
+**Differences in cross-validation procedures**
```

Comment:
I wonder whether the right way of explaining this is no to explain what is measured by a given evaluation procedure.

* A cross-validation procedure is used to measure the risk associated to the procedure of 1) training a model on given data, 2) applying it to new data.

* A train-test split (or train-test-validation split) will measure the risk associated to a given trained model.

(I realize that we could add the distinction above in the model selection section).

In the second case, it makes sense to use an integer seed, as you may want to control fully how the trained model is built. In the first case, it makes little sense.